tracer/src/Datadog.Trace/Util/StringBuilderCache.cs

```diff
@@ -23,7 +23,7 @@ internal static class StringBuilderCache
         [ThreadStatic]
         private static StringBuilder? _cachedInstance;
 
-        public static StringBuilder Acquire(int capacity)
+        public static StringBuilder Acquire(int capacity = MaxBuilderSize)
```

Comment:
Our `StringBuilderCache` type was ~copied from~ based on .NET's internal `StringBuilderCache`, so it's interesting that they use `16` as the default value.

https://github.com/dotnet/runtime/blob/1c10ceecbf5356c33c67f6325072d753707f854e/src/libraries/Common/src/System/Text/StringBuilderCache.cs#L20

Reply:
yes, I wrote about this in the slack thread:

> I'm not sure it makes a huge difference, since we're going to reuse that StringBuilder until it grows bigger that MaxBuilderSize, we might as well allocate it with that size from the get go

Either we needed less space and it gets reused, or we needed more, and in this case no harm in having a big one at first. It's only bad if we create stringbuilders that are too big in short lived threads where they won't get reused.

Where this creates trouble I think is when we use values like 256 that will get over the max allowed capacity when expanded (because the allocated memory doubles on expand, roughly), so if we request a StringBuilder of capacity 256 and we need to write 260 characters to it, it'll get thrown away on release, which is sad.

Which is why I think in our use case, 320 from the beginning is good.

Reply:
Yeah, I was thinking about this as well. We could add telemetry to it, on creation, on reuse and on release, and I think we'd have all the info we need. But I don't think it's something we do much. We do telemetry logs, but not really telemetry metrics ?
Because Microsoft chose 320 based on their experts' advice, but maybe we can choose a value more suited to our use cases. Also, as you say, it may be overthinking, and it may not matter that much in the end :p

Reply:
>  We do telemetry logs, but not really telemetry metrics ?

We have telemetry metrics. Usage is not very high because it's a newer feature and we haven't gone back to add it everywhere in older code.

See https://github.com/DataDog/dd-trace-dotnet/tree/master/tracer/src/Datadog.Trace/Telemetry/Metrics

Reply:
But I think there are issues with using "gauge" metrics (as opposed to "count"). @andrewlock is the expert there.

Reply:
gauge and count metrics are fine; distributions are not

Reply:
I don't see how this use case (`StringBuilder` size) is different from the `Baggage` size we were talking about earlier. I'm not a metrics expert, so I'm probably missing something.

Reply:
🤷‍♂️ it's not different AFAICT 😅 That has the same issues with distributions... Most likely we _want_ a distribution here, not a gauge, so most likely we have the same problem 😄

Reply:
yes we'd probably like to have counters to evaluate the % of "cache miss", but also a distribution of the sizes. I didn't know distribution metrics were causing issues.